src/controllers/controller.bar.js

```diff
@@ -285,12 +290,13 @@ module.exports = DatasetController.extend({
 		var scale = me._getValueScale();
 		var isHorizontal = scale.isHorizontal();
 		var datasets = chart.data.datasets;
-		var value = +scale.getRightValue(datasets[datasetIndex].data[index]);
+		var value = scale._parseValue(datasets[datasetIndex].data[index]);
 		var minBarLength = scale.options.minBarLength;
 		var stacked = scale.options.stacked;
 		var stack = meta.stack;
-		var start = 0;
-		var i, imeta, ivalue, base, head, size;
+		var start = value.start === undefined ? 0 : value.max >= 0 && value.min >= 0 ? value.min : value.max;
```

Comment:
Why do we have to do `value.max >= 0 && value.min >= 0 ? value.min : value.max` instead of just using `value.start`?

Reply:
its because in array the same float-bar can be represented in 2 possible combinations with the difference in direction (top->bottom, bottom->top or left->right, right->left); For example (2,4) and (4,2), its simply the same bar but with different direction. If we will stick to checking only start we will make wrong calculations cause start point will be changing here 2,4 and 4,2. We need to be sure we every time getting the highest value from the pair for positive numbers  and the lowest value for negative numbers to build stacked bars.